### ISSUE TITLE
PCB layout

### ISSUE THREAD

**Issue by dodox1** [sfrwmaker/hakko_t12_stm32#1]
Hello,
do You intend to create/publish PCB for Your project? Can I help with this?

dodo

**Comment by sfrwmaker** [comment]
No, i do not intend to create PCB. I would appreciate if you create it.
The main problem is that the schematics is far from ideal one. First, the DC-DC converter should be changed to another one, 24->3.3v and should provide stable power. Second, the op amp should be changed to SMD variant working at 3.3 volts. Finally,  zener diodes vd5 and vd3 should be removed.
As you can see, there are many modification required in the schematics, and  i think it should be prototyped first.

**Comment by dodox1** [comment]
Did You see this "china" T12 controller board based on STM32 as inspiration?
There is in a few different HW version (2.1S, 2.01 and 2.0)
Reverse engeniered:
https://github.com/jomat/stm32_soldering_iron_controller/tree/master/schematic

**Comment by sfrwmaker** [comment]
Yes, i have one. Cannot tell which version is, but works great. It keeps the preset temperature more accurate than my version, it reaches the temperature faster. I believe, this because of better parameters of PID algorithm. Great, that it has multi-language support. This controller does not use PWM signal to drive the iron, so the schematics is simpler and more compact. Excellent project.
Thank you for the reference.
But this "china" controller has some disadvantages: uses internal clock to show the time (wrongly) and the display is overwhelmed with useless information. The rotary encoder methods are inconvenient: press-and-turn just terrible. There is a red led blinking as iron is powered on.
